instance/models/load_balancer.py

```diff
@@ -196,27 +230,33 @@ def run_playbook(self, ansible_vars):
             self.logger.error("Playbook to reconfigure load-balancing server %s failed.", self)
             raise ReconfigurationFailed
 
-    def _configuration_lock(self):
-        """
-        A Redis lock to protect reconfigurations of this load balancer instance.
-        """
-        return cache.lock("load_balancer_reconfigure:{}".format(self.domain), timeout=900)
-
     def reconfigure(self, triggering_instance_id=None):
         """
         Regenerate the configuration fragments on the load-balancing server.
 
         The triggering_instance_id indicates the id of the instance reference that initiated the
         reconfiguration of the load balancer.
         """
-        self.logger.info("Reconfiguring load-balancing server %s", self.domain)
-        with self._configuration_lock():
+        # This is subject to a race condition, but it doesn't matter:
+        # if two processes increased the version concurrently, the LB would be marked dirty
+        # either way, and only one could successfully grab the lock, while the other would get
+        # kicked out after the blocking timeout.
+        self.configuration_version += 1
+        self.save()
+
+        # Memorize the configuration version, in case new threads change it.
+        with self._configuration_lock(blocking_timeout=1):
```

Comment:
Snap, we missed a major problem here:  This will try to grab the lock, but only block for 1 second in case the lock is held by another process.  However, it will enter the `with` block _regardless of whether it managed to grab the lock_, since there is no conditional code here at all.  I just ran into this issue, and it caused the Cloudera instance (and possibly other instances) to get into a weird state (fixed by reconfiguring the LB again).  I think we need to roll back until we have found a solution here.